great_expectations/execution_engine/sparkdf_execution_engine.py

```diff
@@ -429,7 +441,19 @@ def guess_reader_method_from_path(path: str):
             f"Unable to determine reader method from path: {path}"
         )
 
-    def _get_reader_fn(self, reader, reader_method=None, path=None):
+    @overload
+    def _get_reader_fn(
+        self, reader, reader_method: str = ..., path: Optional[str] = ...
+    ) -> Callable:
+        ...
+
+    @overload
+    def _get_reader_fn(
+        self, reader, reader_method: None = ..., path: str = ...
+    ) -> Callable:
+        ...
```

Comment:
What purpose do these overloads serve?

Reply:
@NathanFarmer These overloads, analogous to those added to `pandas_execution_engine.py` previously, enable us to state what the reader method is as a directive argument to Datasource.add_csv_asset() type methods.